docs/user_guide/preemption.md

```diff
@@ -0,0 +1,249 @@
+
+### PriorityClass
+
+In this example, we will demonstrate the configuration of `yunikorn.apache.org/allow-preemption` using PriorityClass and observe its effect. The default value for this configuration is set to `true`.
+
+```bash
+apiVersion: scheduling.k8s.io/v1
+kind: PriorityClass
+metadata:
+  name: preemption-not-allow
+  annotations:
+    "yunikorn.apache.org/allow-preemption": "false"
+value: 0
+```
+
+We will deploy 8 pods with a resource requirement of 1 to `queue-1`, `queue-2`, and `queue-3`, respectively. We will deploy to `queue-1` and `queue-2` first, followed by a few seconds delay before deploying to `queue-3`. This ensures that the resource usage in `queue-1` and `queue-2` will be greater than that in `queue-3`, depleting all resources in the parent queue and triggering preemption.
+
+| queue                    | Max Resource | Guaranteed Resource | Types of priorities for Pods |
+| ------------------------ | ------------ | ------------------- | ---------------------------- |
+| `priority-class`         | 16           | x                   |                              |
+| `priority-class.queue-1` | 8            | 3                   | allow-preemption             |
+| `priority-class.queue-2` | 8            | 3                   | preemption-not-allow         |
+| `priority-class.queue-3` | 8            | 3                   | allow-preemption             |
+
+Result:
+
+When preemption is triggered, `queue-3` will start searching for a victim. However, since `queue-2` is set with allow-preemption as false, the resources of `queue-1` will be preempted.
+
+It is important to note that setting `yunikorn.apache.org/allow-preemption` is a strong recommendation but not a guarantee. For instance, DaemonSet can still preempt pods even if allow-preemption is set to false.
+
+Furthermore, these rules can be broken if there are no other preemption options available. For example, in this scenario, if the application in `queue-1` has a higher priority than `queue-3`, `queue-2` will be preempted because an application can never preempt another application with a higher priority (law 6), and no other options are available.
+
+| queue                  | Resource before preemption occurs | Resource after preemption occurs       |
+| ---------------------- | --------------------------------- | -------------------------------------- |
+| priority-class.queue-1 | 8 (victim)                        | 5                                      |
+| priority-class.queue-2 | 8                                 | 8                                      |
+| priority-class.queue-3 | 0                                 | 3 (guaranteed resources are satisfied) |
+
+![preemption_priorityclass_case](../assets/preemption_priorityclass_case.png)
+
+### Priority Queue
+
+In addition to utilizing the default PriorityClass in Kubernetes, you can also use [Yunikorn's queue priority](priorities).
```

Comment:
Maybe instead of "you can also use Yunikorn's queue priority", say "you can configure priorities directly on a YuniKorn queue".